File: ios/Classes/ResultCallback.swift
//
//  ResultCallback.swift
//  RCTWuji
//
//  Created by 3 on 2020/12/7.
//

import Foundation

class ResultCallback: NSObject, Callback {
    private var result: FlutterResult?

    init(_ result: FlutterResult?) {
        self.result = result
    }

    func success(_ data: Any?) {
        result?(data)
    }

    func failure(_ code: String, _ message: String) {
        result?(FlutterError.init(code: code, message: message, details: nil))
    }
}

File: android/src/main/kotlin/co/wuji/wuji_rtc_engine/ResultCallback.kt
package co.wuji.wuji_rtc_engine

import co.wuji.rtc.base.Callback
import io.flutter.plugin.common.MethodChannel.Result

class ResultCallback(
        private val result: Result?
) : Callback() {
    override fun success(data: Any?) {
        result?.success(data)
    }

    override fun failure(code: String, message: String) {
        result?.error(code, message, null)
    }
}
